runtime/tokio_util.rs

```diff
@@ -10,15 +51,84 @@ pub fn create_basic_runtime() -> tokio::runtime::Runtime {
     // The default value is 512, which is an unhelpfully large thread pool. We
     // don't ever want to have more than a couple dozen threads.
     .max_blocking_threads(32)
+    .global_queue_interval(40960)
+    .event_interval(10000)
     .build()
     .unwrap()
 }
 
+pub fn run_local2<F, R>(future: F) -> R
+where
+  F: std::future::Future<Output = R> + 'static,
+  F::Output: Send + 'static,
+{
+  let local = LocalRuntime::new();
+
+  local.spawn(async move {
+    let handle = tokio::runtime::Handle::current();
+    let runtime_monitor = tokio_metrics::RuntimeMonitor::new(&handle);
+
+    // print runtime metrics every 500ms
+    let frequency = std::time::Duration::from_millis(1_000);
+    tokio::spawn(async move {
+      for metrics in runtime_monitor.intervals() {
+        println!("{:#?}", metrics);
+        tokio::time::sleep(frequency).await;
+      }
+    });
+  });
+
+  let join_handle =
+    local.spawn(unsafe { make_me_send::MakeMeSend::new(future) });
+  local.block_on(async move { join_handle.await }).unwrap()
+}
+
+pub fn run_local3<F>(future: F)
+where
+  F: std::future::Future + 'static,
+  F::Output: Send + 'static,
+{
+  let local = LocalRuntime::new();
+  local.spawn(unsafe { make_me_send::MakeMeSend::new(future) });
+}
+
 pub fn run_local<F, R>(future: F) -> R
 where
   F: std::future::Future<Output = R>,
 {
-  let rt = create_basic_runtime();
-  let local = tokio::task::LocalSet::new();
-  local.block_on(&rt, future)
+  let local = LocalRuntime::new();
+  local.block_on(future)
+}
+
+pub mod make_me_send {
+  use core::future::Future;
+  use core::pin::Pin;
+  use core::task::Context;
+  use core::task::Poll;
+
+  pub struct MakeMeSend<F> {
+    future: F,
+  }
+
+  impl<F> MakeMeSend<F> {
+    /// SAFETY: You must ensure that the future is not used
+    /// on the wrong thread.
+    pub unsafe fn new(future: F) -> Self {
+      Self { future }
+    }
+  }
+
+  unsafe impl<F> Send for MakeMeSend<F> {}
+
+  impl<F: Future> Future for MakeMeSend<F> {
+    type Output = F::Output;
+
+    fn poll(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<F::Output> {
+      unsafe {
+        let me = Pin::into_inner_unchecked(self);
+        let future = Pin::new_unchecked(&mut me.future);
+        future.poll(cx)
+      }
+    }
+  }
```

Comment:
This was also done by @Darksonn; it's to cheat Tokio so that we can `spawn()` a future instead of `spawn_local()` which requires that we are running inside `LocalSet`. This hack is fine because we use a current thread runtime (but obviously it will wreak havoc if someone tries to use it with multi-threaded flavor). 

This should also be moved to `deno_core` and maybe get a better name